source/extensions/transport_sockets/tls/context_manager_impl.cc

```diff
@@ -24,30 +24,44 @@ void ContextManagerImpl::removeEmptyContexts() {
   contexts_.remove_if([](const std::weak_ptr<Envoy::Ssl::Context>& n) { return n.expired(); });
 }
 
+void ContextManagerImpl::removeOldContext(std::shared_ptr<Envoy::Ssl::Context> old_context) {
+  if (old_context) {
+    contexts_.remove_if([old_context](const std::weak_ptr<Envoy::Ssl::Context>& n) {
+      std::shared_ptr<Envoy::Ssl::Context> sp = n.lock();
+      if (old_context && sp) {
```

Comment:
Isn't `old_context` here redundant as it's checked in L28?

Reply:
Yeah. It is redundant